community/kernel/src/main/java/org/neo4j/kernel/api/index/IndexAccessor.java

```diff
@@ -90,6 +91,17 @@
      */
     ResourceIterator<File> snapshotFiles() throws IOException;
 
+    /**
+     * Verifies that each value in this index is unique.
+     * Index is guaranteed to not change while this call executes.
+     *
+     * @param propertyAccessor {@link PropertyAccessor} for accessing properties from database storage
+     * in the event of conflicting values.
+     * @throws IndexEntryConflictException for first detected uniqueness conflict, if any.
+     * @throws IOException on error reading from source files.
+     */
+    void verifyDeferredConstraints( PropertyAccessor propertyAccessor ) throws IndexEntryConflictException, IOException;
```

Comment:
Hmm, it seems that a LuceneIndexAccessor doesn't quite know whether it's a uniqueness index or not. Would it be useful to pass it in only to check that and throw exception if that's the case? I don't feel it would add much, would it?